apps/passport-server/test/lemonade/MockLemonadeServer.ts

```diff
@@ -0,0 +1,110 @@
+import { loadDevMessages, loadErrorMessages } from "@apollo/client/dev";
+import { stringify } from "csv-stringify/sync";
+import { MockedRequest, RequestHandler, graphql, rest } from "msw";
+import urljoin from "url-join";
+import {
+  LemonadeTicket,
+  LemonadeTicketType
+} from "../../src/apis/lemonade/lemonadeAPI";
+import { LemonadeDataMocker } from "./LemonadeDataMocker";
+
+// Apollo client doesn't load error messages by default so we have to call this
+loadDevMessages();
+loadErrorMessages();
+
+export function getMockLemonadeHandlers(
+  mocker: LemonadeDataMocker,
+  backendUrl: string
+): RequestHandler[] {
+  const handlers = [];
+
+  // In the real API, the authorization token is an OAuth token, which maps
+  // back to the client ID. For testing purposes, we just send the client ID
+  // as the token, to avoid having to mock out the whole OAuth flow.
+  // TODO actually mock this with a bit more fidelity, so we can mock token expiry
+  const checkClientId = (
+    mocker: LemonadeDataMocker,
+    req: MockedRequest
+  ): string => {
+    const clientId = req.headers.get("Authorization")?.split(" ")[1];
+    if (!clientId || !mocker.getAccount(clientId)) {
+      throw new Error(`Invalid client ID: ${clientId}`);
+    }
+    return clientId;
+  };
+
+  // Mock GraphQL endpoints
```

Comment:
amazing

Reply:
This was much less difficult than I thought it would be